pkg/registry/core/pod/rest/subresources.go

```diff
@@ -113,7 +115,24 @@ func (r *AttachREST) Connect(ctx context.Context, name string, opts runtime.Obje
 	if err != nil {
 		return nil, err
 	}
-	return newThrottledUpgradeAwareProxyHandler(location, transport, false, true, responder), nil
+	handler := proxy.NewUpgradeAwareHandler(location, transport, false, true, proxy.NewErrorResponder(responder))
+	handler.MaxBytesPerSec = capabilities.Get().PerConnectionBandwidthLimitBytesPerSec
+	if utilfeature.DefaultFeatureGate.Enabled(features.ClientRemoteCommandWebsockets) {
+		// Wrap the upgrade aware handler to implement stream translation
+		// for WebSocket/V5 upgrade requests.
+		streamOptions := proxy.Options{
+			Stdin:  attachOpts.Stdin,
+			Stdout: attachOpts.Stdout,
+			Stderr: attachOpts.Stderr,
+			Tty:    attachOpts.TTY,
+		}
+		if err := proxy.ValidateOptions(streamOptions); err != nil {
+			return nil, err
+		}
```

Comment:
doing this validation at this layer in the API server looks new... I see similar validation in the kubelet, but shouldn't we pass through the options and let the kubelet reject the request if the options are invalid? I'm not sure we can assume those combinations will always be invalid

Reply:
Removed this validation.